hydra-node/exe/hydra-node/Main.hs

```diff
@@ -86,15 +94,41 @@ main = do
         pparams <- case toLedgerPParams ShelleyBasedEraBabbage protocolParams of
           Left err -> throwIO (ConfigurationParseException err)
           Right bpparams -> pure bpparams
-        withCardanoLedger chainConfig pparams $ \ledger -> do
-          persistence <- createPersistenceIncremental $ persistenceDir <> "/state"
+        let onlineOrOfflineConfig = case offlineConfig of
+              Nothing -> Right chainConfig
+              Just offlineConfig' -> Left offlineConfig'
+        
+        let DirectChainConfig{networkId, nodeSocket} = chainConfig
+
+        globals <- case offlineConfig of
+              Nothing -> do  --online
+                globals' <- newGlobals =<< queryGenesisParameters networkId nodeSocket QueryTip
+                pure globals'
+              Just OfflineConfig{ledgerGenesisFile} -> do  --offline
+                loadGlobalsFromGenesis ledgerGenesisFile
+
+        withCardanoLedger pparams globals $ \ledger -> do
+          persistence <- createStateChangePersistence (persistenceDir <> "/state") (offlineOptionsNormalizedUtxoWriteBackFilePath =<< leftToMaybe onlineOrOfflineConfig)
           (hs, chainStateHistory) <- loadState (contramap Node tracer) persistence initialChainState
+          
+          let headId = HeadId "HeadId"
+          case offlineConfig of
+            Nothing -> pure ()
+            Just (OfflineConfig {initialUTxOFile})-> do
+              initialUTxO :: UTxOType Tx <- readJsonFileThrow (parseJSON @(UTxOType Tx)) initialUTxOFile
+              initializeStateIfOffline chainStateHistory initialUTxO headId party contestationPeriod (putEvent . OnChainEvent)
```

Comment:
I think this (yielding of constant observations to simulate a head opening) can be moved to `withOfflineChain` directly, which would simplify the main function (less casing).